### PR TITLE
Working Operator

### PR DESCRIPTION
**Pull request by jmprusi**
TODO

- [x] Move plans to CRD instead of Swagger
- [x] Push changes to production
- [x] Allow setting a default plan
- [x] BUG: When the API def is removed, it's not removed from System.
- [x] Create service based on CR name if it doesn't exist.
- [x] Set private URL based on the CR

### CODE REVIEW

**Comment by PhilipGough** [comment]
@jmprusi I've investigated the task  `BUG: When the API def is removed, it's not removed from System.`

The code on this branch does a delete of a service and I have observed that the service is deleted so it is intermittent at best. I actually havent seen it not being removed. https://github.com/3scale/ostia/blob/e2e-tests/threescaleAPI-operator/pkg/threescaleAPI/reconciliation_e2e_test.go#L28-L40

I spoke to @Martouta and there is an open issue around this. See https://github.com/3scale/porta/pull/6 and https://issues.jboss.org/browse/THREESCALE-1194?focusedCommentId=13626316&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13626316

To summarise a event is kicked off to mark a service for deletion and if everything goes well we are returned a `200` but the background job that should be triggered may not run in some circumstances.

Not sure there is much we can do from our side in relation to this.

**Comment by jmprusi** [comment]
@PhilipGough Good to know.

I assume the service creation is synchronous...

**Comment by jmprusi** [comment]
@PhilipGough should we merge this into the backend-poc branch?

**Comment by PhilipGough** [comment]
@jmprusi I'm ok with that if you are not worried about the failing tests. I haven't had a chance to look yet to be honest

**Comment by jmprusi** [comment]
@PhilipGough those are not actually related to this operator.. (we need to fix that) .

I will just go ahead and merge it!
